openmls/src/group/tests/kat_transcript_hashes.rs

```diff
@@ -0,0 +1,284 @@
+//! # Known Answer Tests for the transcript hashes
+//!
+//! See <https://github.com/mlswg/mls-implementations/blob/master/test-vectors.md>
+//! for more description on the test vectors.
+
+use std::convert::TryFrom;
+
+use openmls_rust_crypto::OpenMlsRustCrypto;
+use openmls_traits::{crypto::OpenMlsCrypto, random::OpenMlsRand, OpenMlsCryptoProvider};
+use serde::{self, Deserialize, Serialize};
+use thiserror::Error;
+use tls_codec::Serialize as TlsSerializeTrait;
+
+#[cfg(test)]
+use crate::test_utils::{read, write};
+use crate::{
+    binary_tree::array_representation::LeafNodeIndex,
+    framing::{mls_auth_content::AuthenticatedContent, *},
+    group::{
+        tests::utils::{generate_credential_bundle, randombytes},
+        *,
+    },
+    messages::*,
+    schedule::*,
+    test_utils::*,
+    versions::ProtocolVersion,
+};
+
+const TEST_VECTOR_PATH_READ: &str = "test_vectors/transcript-hashes.json";
+const TEST_VECTOR_PATH_WRITE: &str = "test_vectors/transcript-hashes-new.json";
+const NUM_TESTS: usize = 100;
+
+/// ```json
+/// {
+///   "cipher_suite": /* uint16 */,
+///
+///   /* Chosen by the generator */
+///   "confirmation_key": /* hex-encoded binary data */,
+///   "authenticated_content": /* hex-encoded TLS serialized AuthenticatedContent */,
+///   "interim_transcript_hash_before": /* hex-encoded binary data */,
+///
+///   /* Computed values */
+///   "confirmed_transcript_hash_after": /* hex-encoded binary data */,
+///   "interim_transcript_hash_after": /* hex-encoded binary data */,
+/// }
+/// ```
+#[derive(Serialize, Deserialize, Debug, Clone, Default)]
+pub struct TranscriptTestVector {
+    pub cipher_suite: u16,
+
+    #[serde(with = "hex::serde")]
+    pub confirmation_key: Vec<u8>,
+    #[serde(with = "hex::serde")]
+    pub authenticated_content: Vec<u8>,
+    #[serde(with = "hex::serde")]
+    pub interim_transcript_hash_before: Vec<u8>,
+
+    #[serde(with = "hex::serde")]
+    pub confirmed_transcript_hash_after: Vec<u8>,
+    #[serde(with = "hex::serde")]
+    pub interim_transcript_hash_after: Vec<u8>,
+}
+
+/// Transcript test vector error
+#[derive(Error, Debug, PartialEq, Eq, Clone)]
+pub enum TranscriptTestVectorError {
```

Comment:
We don't really need the errors anymore because the interop runner doesn't do test vectors anymore.